mobilecoind/api/proto/mobilecoind_api.proto

```diff
@@ -128,6 +128,9 @@ message TxProposal {
     /// This is needed to map recipients to their respective TxOuts.
     map<uint64, uint64> outlay_index_to_tx_out_index = 5;
 
+    /// A list of the confirmation numbers, in the same order
+    /// as the outlays.
+    repeated bytes outlay_confirmation_numbers = 6;
```

Comment:
alternatively this could be put inside of `Outlay` - I think this might simplify a few things.

Reply:
Outlays are created by the caller and are immutable after that, so they can't be modified. You'll notice that it also might have made sense to put the generate TXOs on the outlays but instead a mapping table was used for the same reason.